sdk/core/core/inc/az_http_transport.h

```diff
@@ -89,24 +89,76 @@ az_http_request_get_header(_az_http_request const* request, int32_t index, az_pa
  * call this function like below to get only the http method and ignore getting url and body.
  *   `az_http_request_get_parts(request, &method, NULL, NULL)`
  *
- * This function is expected to be used by transport layer only.
+ * @remarks This function is expected to be used by transport layer only.
  *
- * @param request HTTP request to get parts from.
+ * @param[in] request HTTP request to get parts from.
  * @param[out] out_method __[nullable]__ Pointer to write HTTP method to. Use `NULL` to ignore
  * getting this value.
  * @param[out] out_url __[nullable]__ Pointer to write URL to. Use `NULL` to ignore getting this
  * value.
  * @param[out] out_body __[nullable]__ Pointer to write HTTP request body to. Use `NULL` to ignore
  * getting this value.
  *
- * @retval AZ_OK Success.
+ * @retval An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
  */
 AZ_NODISCARD az_result az_http_request_get_parts(
     _az_http_request const* request,
     az_http_method* out_method,
     az_span* out_url,
     az_span* out_body);
 
+/**
+ * @brief This function is expected to be used by transport adapters like curl. Use it to write
+ * content from \p write to \p response.
+ *
+ * @remarks This is a convenient way of hiding the internal implementation of az_htt_response.
```

Comment:
Is this useful/relevant for the customer to know? I think we should remove this remark (you can make it a regular comment that doesn't show up as part of docs, if you think it s important for us to know as implementors).

```suggestion
 * @remarks This is a convenient way of hiding the internal implementation of az_http_response.
```